aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUserPool.java

```diff
@@ -268,6 +273,12 @@ public CognitoUserPool(Context context, String userPoolId, String clientId, Stri
     public CognitoUserPool(Context context, String userPoolId, String clientId, String clientSecret, ClientConfiguration clientConfiguration, Regions region, String pinpointAppId) {
         initialize(context);
         this.context = context;
+        if (userPoolId.isEmpty() || clientId.isEmpty()) {
```

Comment:
Can userPoolId or clientId be null?

Reply:
Although they are not set to @NonNull, there's rarely the case to pass a null to userPoolId or or clientId. In real scenario, they are parsed from `amplifyconfiguration.json` file. This check is to ensure they're not empty strings.